src/core/parseutil.cpp

```diff
@@ -51,6 +58,10 @@ QString ParseUtil::createErrorMessage(const QString &message, const QString &exp
 }
 
 QString ParseUtil::readTextFile(const QString &path) {
+    if (fileCache.contains(path)) {
+        return fileCache[path];
+    }
```

Comment:
This introduces a bug when `readTextFile` is used on a file that has changed, because it will instead return the file in its original state. Example: Open the palette editor, change a color, save and exit the palette/tileset editor, change to a different tileset, change back to the previous tileset, re-open the palette editor, the color change has been lost